bookkeeper-server/src/main/java/org/apache/bookkeeper/util/collections/ConcurrentLongLongPairHashMap.java

```diff
@@ -220,22 +329,36 @@ public void forEach(BiConsumerLongPair processor) {
     }
 
     // A section is a portion of the hash map that is covered by a single
+    @Setter
     @SuppressWarnings("serial")
     private static final class Section extends StampedLock {
         // Keys and values are stored interleaved in the table array
         private volatile long[] table;
 
         private volatile int capacity;
         private volatile int size;
-        private int usedBuckets;
-        private int resizeThreshold;
-
-        Section(int capacity) {
+        private volatile int usedBuckets;
+        private int resizeThresholdUp;
+        private int resizeThresholdBelow;
+        private volatile float mapFillFactor;
```

Comment:
Why do we want these to be modifiable at runtime?
I cannot find a good usecase.

We should have a smaller API, please remember that every public method is 'public API' and we will have to maintain it I'm the future

Reply:
@eolivelli  
The purpose of this PR is to solve the problem of pendingAcks taking up too much memory in pulsar.
I would like to support dynamic configuration of these parameters in pulsar for the following reasons:
1. If the current memory resources are relatively tight, we need to increase the mapIdleFactor to reduce the wasted memory space, but at this time, the number of rehashes will increase, and the time-consuming of put and remove will increase.
2. If we want put/remove to take less time, we need to reduce mapIdleFactor, reduce the number of rehash, and exchange space for time;

Reply:
It's a trade-off strategy between time and space。

Reply:
For another example, during runtime, the consumption of memory resources increases due to a brief increase in traffic. At this time, we can temporarily adjust these parameters to release some free memory resources.

Reply:
I think it would be a bit tricky to auto-adjust at runtime. 

For now we could just leave it to configure at creation time. I agree that for cases like pending acks we need to be aggressive in reclaiming space but I don't see how to change that based on load.

Reply:
> I think it would be a bit tricky to auto-adjust at runtime.

1. Similar to other dynamic configurations of pulsar, the configuration can be adjusted without restarting the broker.

> For now we could just leave it to configure at creation time. I agree that for cases like pending acks we need to be aggressive in reclaiming space but I don't see how to change that based on load.

2. For example, through the size() and capacity() methods, you can calculate the memory usage percentage usePercent=size()/capacity(), so as to determine whether the configuration parameters need to be adjusted;

Reply:
1. Reducing mapFillFactor and mapIdleFactor can allocate more memory space, reduce the number of rehash, and reduce the time-consuming of put and remove methods;
2. Adding mapFillFactor and mapIdleFactor can improve the memory space usage, but the number of rehash will increase, and the time-consuming of put and remove methods will also increase;

Reply:
If you think adding these interfaces is unreasonable, I can remove them: @merlimat  @eolivelli 
①setMapFillFactor
②setMapIdleFactor
③setExpandFactor
④setShrinkFactor
⑤setAutoShrink

Reply:
Fixed @eolivelli @merlimat